Add Teleport Documentation (#2004)
giantswarm/docs: src/content/platform-overview/security/cluster-security/cluster-access/index.md

The Kubernetes API of each workload cluster is exposed to customers. Authorized users of the cluster can be managed by connecting an external identity provider to the Kubernetes API, or by provisioning them directly through the Giant Swarm REST API. In either case, you retain full control over user management.

## Admin access via Teleport

### What is Teleport?

[Teleport](https://goteleport.com/) is an open-source solution for managing secure access to infrastructure using identity-aware reverse proxy and short-lived certificates instead of passwords or long-lived keys. This helps strengthen security and simplifies compliance with regulations and network topology.

> Some of the advantages of Teleport:

Reviewer: ```suggestion
Some advantages of Teleport:
```

- Simplify VPN architecture
- Support more network layouts and overcome CIDR range limitations
- Increase security
- Audit logging
- Access to audit logs for customers

### Teleport at Giant Swarm

We use Teleport as a standard solution to maintain access to managed infrastructure for operational needs. Teleport offers us with robust audit and access logs which we lacked in our previous access management solution. Teleport's identity-aware reverse proxy with TLS routing simplifies and secures network access requirements, as VPN is no longer needed nor any additional ports needs to be open at customer network, as such, Teleport works seamlessly behind corporate firewalls, where only outbound HTTPS traffic to Giant Swarm hosted Teleport cluster needs to be allowed.

### Teleport secured access points

- **SSH** - SSH access is secured with Teleport (Node Agent).

- **Kubernetes API** - Usage of the Kubernetes API on the management/workload cluster is also secured with Teleport (Kube Agent).

The following diagram shows our Teleport architecture in more detail:

![Teleport Architecture](teleport.png)

> Teleport cluster access is based on GitHub SSO with MFA. Only users in the GitHub Giant Swarm Organization are allowed to authenticate.

Reviewer: Security: Why mention which SSO provider we use?

Reviewer: Same for the graphics (GitHub SSO logo)? (Maybe use something generic?)

Author: I used the GitHub SSO as it was previously described here as well:
- https://docs.giantswarm.io/platform-overview/security/cluster-security/cluster-access/#vpn-secured-access-points

Also, another reason is because we run teleport open-source version which only supports GitHub SSO.

Reviewer: OK! No problem then, I guess.

> We run highly resilient and available Teleport cluster on AWS with robust access and audit logs for each user session.

Reviewer: Security: Why mention on which cloud provider we use to host our Teleport infrastructure?

Reviewer: Same for the graphics (AWS logo)?

> We run highly resilient and available Teleport cluster on AWS with robust access and audit logs for each user session.

Reviewer: ```suggestion
We run a highly resilient and available Teleport cluster with robust access and audit logs for each user session.
```
